src/nxt_conf_validation.c

```diff
@@ -1250,107 +1358,235 @@ nxt_conf_vldt_array_iterator(nxt_conf_validation_t *vldt,
+    }
 
-    length = name.length + 1;
-    length = nxt_min(length, sizeof(string));
+    if (nxt_conf_type(value) != NXT_CONF_STRING) {
+        return nxt_conf_vldt_error(vldt, "The \"%V\" environment value must be "
+                                   "a string.", name);
+    }
 
-    nxt_cpystrn((u_char *) string, name.start, length);
+    nxt_conf_get_string(value, &str);
 
-    return validator(vldt, string);
+    if (nxt_memchr(str.start, '\0', str.length) != NULL) {
+        return nxt_conf_vldt_error(vldt, "The \"%V\" environment value must "
+                                   "not contain null character.", name);
+    }
+
+    return NXT_OK;
 }
 
+typedef struct {
+    u_char usr;
+    u_char mnt;
+    u_char net;
+    u_char ipc;
+    u_char uts;
+    u_char pid;
+    u_char cgr;
+} nxt_conf_vldt_clone_namespaces_conf_t;
+
+static nxt_conf_map_t nxt_conf_vldt_clone_namespaces_conf_map[] = {
+    {
+        nxt_string("credential"),
+        NXT_CONF_MAP_INT8,
+        offsetof(nxt_conf_vldt_clone_namespaces_conf_t, usr),
+    },
+
+    {
+        nxt_string("mount"),
+        NXT_CONF_MAP_INT8,
+        offsetof(nxt_conf_vldt_clone_namespaces_conf_t, mnt),
+    },
+
+    {
+        nxt_string("network"),
+        NXT_CONF_MAP_INT8,
+        offsetof(nxt_conf_vldt_clone_namespaces_conf_t, net),
+    },
+
+    {
+        nxt_string("ipc"),
+        NXT_CONF_MAP_INT8,
+        offsetof(nxt_conf_vldt_clone_namespaces_conf_t, ipc),
+    },
+
+    {
+        nxt_string("uname"),
+        NXT_CONF_MAP_INT8,
+        offsetof(nxt_conf_vldt_clone_namespaces_conf_t, uts),
+    },
+
+    {
+        nxt_string("pid"),
+        NXT_CONF_MAP_INT8,
+        offsetof(nxt_conf_vldt_clone_namespaces_conf_t, pid),
+    },
+
+    {
+        nxt_string("cgroup"),
+        NXT_CONF_MAP_INT8,
+        offsetof(nxt_conf_vldt_clone_namespaces_conf_t, cgr),
+    },
+};
 
 static nxt_int_t
-nxt_conf_vldt_user(nxt_conf_validation_t *vldt, char *user)
+nxt_conf_vldt_clone_namespaces(nxt_conf_validation_t *vldt, nxt_conf_value_t *value,
+    void *data)
 {
-    struct passwd  *pwd;
+    nxt_conf_vldt_clone_namespaces_conf_t ns;
+    nxt_int_t                             ret;
 
-    nxt_errno = 0;
+    ret = nxt_conf_vldt_object(vldt, value, data);
+    if (ret != NXT_OK) {
+        return ret;
+    }
 
-    pwd = getpwnam(user);
+    nxt_memset(&ns, 0, sizeof(nxt_conf_vldt_clone_namespaces_conf_t));
 
-    if (pwd != NULL) {
-        return NXT_OK;
+    ret = nxt_conf_map_object(vldt->pool, value,
+                              nxt_conf_vldt_clone_namespaces_conf_map,
+                              nxt_nitems(nxt_conf_vldt_clone_namespaces_conf_map),
+                              &ns);
+    if (ret != NXT_OK) {
+        return ret;
     }
 
-    if (nxt_errno == 0) {
-        return nxt_conf_vldt_error(vldt, "User \"%s\" is not found.", user);
+    /* we can allow this when proxy feature arrives */
+    if (ns.ipc) {
```

Comment:
If we don't support it currently, then it's better to not provide it as a possible setting. Even when proxy feature will arrive, it may require more configuration than just turning `ipc` capability on.

Anyway, such check should be made in the validation handler in https://github.com/nginx/unit/pull/289/files#diff-48026d2a45c1225dc17bae6d8287531cR391